oauth2client/service_account.py

```diff
@@ -461,3 +465,94 @@ def create_delegated(self, sub):
         result._private_key_pkcs12 = self._private_key_pkcs12
         result._private_key_password = self._private_key_password
         return result
+
+
+def _datetime_to_secs(utc_time):
+   # time_delta.total_seconds() not supported in Python 2.6
+   epoch = datetime.datetime(1970, 1, 1)
+   time_delta = utc_time - epoch
+   return time_delta.days * 86400 + time_delta.seconds
+
+
+class JWTAccessCredentials(GoogleCredentials):
+    """Self signed JWT credentials.
+
+    Makes an assertion to server using a self signed JWT from service account
+    credentials.  These credentials do NOT use OAuth 2.0 and instead authenticate
+    directly.
+    """
+    MAX_TOKEN_LIFETIME_SECS = 3600
```

Comment:
AFAICT the stigma towards class-scope attributes stem from them being overridable (or surprisingly not) in subclasses/instances-in-a-way, making things confusing in the long run by paradoxically breaking encapsulation. Or that they represent a 'second' way of doing things. Or... other style reasons.

Kind of a silly thought given how we've all been taught, "Encapsulation! Scopes! Classes!" since day one, but... ¯\_(ツ)_/¯